### PR TITLE
Update pagination component

### PR DESCRIPTION
**Pull request by ryaplots**
#### Summary

This PR updates the existent pagination component according to the new design

#### Changes

- Update styling
- Add page size select
- Add page input

Screenshot 2024-03-06 at 10 02 35

This still needs the following:
1. Decide on the steps of the page size select

- To test run the stack on staging

#### Checklist

- [ ] Scope: The referenced issue is addressed, there are no unrelated changes.
- [ ] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] The steps/process to test this feature are clearly explained including testing for regressions.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [ ] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Comment by kschiffer** [comment]
I don't really see an appearance like in your screenshot:
image
- Different height of the page buttons
- Different font-weight

Also the drop-ups are hidden by the table:
image

**Comment by kschiffer** [comment]
Available page sizes should be:
20 (default), 30, 50, 100

Please use `useCookieState` to store these for now, so that the selection persists between reloads.

We could enable page sizes of up to 1000 but there are some considerations that make me reluctant with this:
- 1k page sizes might put a significant load on the backend
- Applications to per-item fetches of their end device counts so big page sizes would result in many requests that could potentially result in hitting rate-limits

@adriansmares any thoughts on this?

**Review by kschiffer — CHANGES_REQUESTED** [review]
See above.

**Comment by ryaplots** [comment]
Updated the screenshot and tested on staging

File: pkg/webui/containers/fetch-table/index.js [hunk 1]
```diff
@@ -304,6 +310,8 @@ const FetchTable = props => {
           page={page}
           totalCount={totalCount}
           pageSize={pageSize}
+          setPageSize={setPageSize}
+          setPageSizeCookie={setPageSizeCookie}
```
Comment (kschiffer) [inline pkg/webui/containers/fetch-table/index.js]: There should be just one `setPageSize` handler which uses the cookie state. There shouldn't be the need to have an additional state variable for it.

**Review by kschiffer — CHANGES_REQUESTED** [review]
One thing I realized is that different page size settings will make the URLs less deterministic since the exact results that are shown will be dependent on the page sizes. I don't really see a good way to address this at the moment, so I believe we have to live with this. One could add the page size to the URL search params but that could then be in conflict with the user preference.

**Comment by ryaplots** [comment]
> One thing I realized is that different page size settings will make the URLs less deterministic since the exact results that are shown will be dependent on the page sizes. I don't really see a good way to address this at the moment, so I believe we have to live with this. One could add the page size to the URL search params but that could then be in conflict with the user preference.

We could add the page size to the url based on the cookie, since the cookie is available everywhere if it is set. So, in the link would look like `/applications?custom-page-size=${pageSizeCookie}` or `/applications` in case there is no cookie set.

**Comment by kschiffer** [comment]
Yeah, thinking about it, I believe the following could work:
- When navigating to a table organically in the Console, automatically add the `page-size` to the URL based on the current cookie (or fallback to the default)
  - This way, when sharing table result pages, the page size will always be included 
- When navigating to a table from a URL that has the `page-size` search param present -> use it instead of the cookie
- Only update the cookie when actively selecting a different page size

This way, for the time the FetchTable is rendered with a page size other than the currently set, it should use it, but when navigating away and back to a fetch table, the cookie-based page size should be used again

**Review by kschiffer — CHANGES_REQUESTED** [review]
See above. Otherwise LGTM.

**Review by kschiffer — CHANGES_REQUESTED** [review]
Doesn't seem to work as expected. While changing the page size manually (via the select input) works, manipulating the page through the query parameter does not work correctly.
The page size will show the amount entered, but the number of items displayed and the number of pages is unaffected and stays with the cookie value.

File: pkg/webui/components/pagination/index.js [hunk 2]
```diff
@@ -30,46 +43,112 @@ const Pagination = ({
   marginPagesDisplayed,
   hideIfOnlyOnePage,
   pageCount,
+  pageSize: propPageSize,
+  setPageSize,
   ...rest
 }) => {
+  const [selectedPage, setSelectedPage] = useState(forcePage)
+  const [pageSize, setQueryPageSize] = useQueryState('page-size', propPageSize)
```
Comment (kschiffer) [inline pkg/webui/components/pagination/index.js:57]: It's important to validate the page size as well and make sure that only values as provided in the select box are allowed.

File: pkg/webui/components/pagination/index.js [hunk 3]
```diff
@@ -12,24 +12,29 @@
...
 
+import getCookie from '@console/lib/table-utils'
+
 import Select from '../select'
 import Input from '../input'
 import Button from '../button'
 
 import style from './pagination.styl'
 
+const allowedPageSizes = [20, 30, 50, 100]
```
Comment (kschiffer) [inline pkg/webui/components/pagination/index.js]: Nit: these should be put into a constant module.

**Review by kschiffer — APPROVED** [review]
Works now, thanks!

**Comment by kschiffer** [comment]
image
Also the "Go to page" input is now longer somehow.

**Comment by ryaplots** [comment]
>  Also the "Go to page" input is now longer somehow.

On what page did you see it? I cannot reproduce. @kschiffer

**Comment by kschiffer** [comment]
I've cleaned and rebuilt everything and now the issue is gone. Seems like some hickup on my side.

**Pull request merged**